### PR TITLE
feat(ICU): Upgrade to 58.2+, restructure exclude dirs, add FreeBSD Tools

### PR DESCRIPTION
**Pull request by fsbruva**
I set out to fix the icudt*.dat issue, but couldn't help myself from fixing the rest of the stuff.

### CODE REVIEW

File: buildme.pl [hunk 1]
```diff
@@ -25,21 +25,34 @@
...
+my $icuLittleEnd = "icudt[0-9][0-9]l.dat ";
+my $freebsdDirs = "i386-freebsd-64int ";
+my $solarisDirs = "i86pc-solaris-thread-multi-64int ";
+my $armLinuxDirs = "arm-linux armhf-linux ";
+my $sparcLinuxDirs = "sparc-linux sparc-unknown-linux ";
+my $mainLinuxDirs = "i386-linux x86_64-linux "
+my $linuxDirs = $armLinuxDirs . $sparcLinuxDirs . $mainLinuxDirs . "powerpc-linux ";
+my $win32Dirs = "MSWin32-x86 ";
+my $macOSDirs = "darwin darwin-x86_64 darwin-thread-multi ";
+my $nonLinuxOSDirs = $freebsdDirs . $win32Dirs . $solarisDirs;
```
Comment (mherger) [inline buildme.pl]: $macOSDirs is missing here

File: buildme.pl [hunk 2]
```diff
@@ -25,21 +25,34 @@
...
 my $windowsPerlPath = "$windowsPerlDir\\bin\\perl.exe";
 
+my $icuBigEnd = "icudt[0-9][0-9]b.dat ";
+my $icuLittleEnd = "icudt[0-9][0-9]l.dat ";
+my $freebsdDirs = "i386-freebsd-64int ";
+my $solarisDirs = "i86pc-solaris-thread-multi-64int ";
+my $armLinuxDirs = "arm-linux armhf-linux ";
+my $sparcLinuxDirs = "sparc-linux sparc-unknown-linux ";
+my $mainLinuxDirs = "i386-linux x86_64-linux "
+my $linuxDirs = $armLinuxDirs . $sparcLinuxDirs . $mainLinuxDirs . "powerpc-linux ";
```
Comment (mherger) [inline buildme.pl]: If already you move everything to variables, why didn't you do the same for powerpc-linux?

File: buildme.pl (hunk 1 above)
Comment (fsbruva) [reply buildme.pl]: Right you are. Will fix.

File: buildme.pl (hunk 2 above)
Comment (fsbruva) [reply buildme.pl]: Probably a good point - but my initial goal was to compress the information by combining frequently coupled data, so it was easier to read and maintain. Now I see that if there is a sudden resurgence of PowerPC's that use Perl with 64int or thread or something, we ought to be flexible. Will fix.

**Comment by fsbruva** [comment]
Fixed.

**Comment by mherger** [comment]
Oops... sorry, should probably have used my latest build on aarch64 to test your change :-(

**Comment by fsbruva** [comment]
More fixes/suggestions for me?

**Comment by mherger** [comment]
@fsbruva - as I mentioned in my previous comment I did commit a change which broke your pull request. Could you please rebase it?

**Comment by fsbruva** [comment]
I saw the commit after I asked. Yes I can rebase.

**Comment by fsbruva** [comment]
@mherger Rebase/squash is complete. Also fixed shebang and incremented version.

File: buildme.pl [hunk 3]
```diff
@@ -17,29 +17,50 @@
...
+my $freebsdDirs = "i386-freebsd-64int ";
+my $solarisDirs = "i86pc-solaris-thread-multi-64int ";
+my $mainLinuxDirs = "i386-linux x86_64-linux ";
+my $armHFLinuxDirs = "armhf-linux ";
+my $arm64LinuxDirs = "aarch64-linux ";
+my $armLinuxDirs = "arm-linux aarch-linux " . $armHFLinuxDirs . $arm64LinuxDirs;
+my $sparcLinuxDirs = "sparc-linux sparc-unknown-linux ";
+my $ppcLinuxDirs = "powerpc-linux ";
+my $linuxDirs = $armLinuxDirs . $sparcLinuxDirs . $mainLinuxDirs . $ppcLinuxDirs;
+my $win32Dirs = "MSWin32-x86 ";
```
Comment (mherger) [inline buildme.pl]: That should be MSWin32-x86-multi-thread.

**Review by mherger — COMMENTED** [review]
After trying to review this change I don't think it comes with a real advantage. I do see the idea of having items in groups which can be re-used. But trying to understand which configuration would be using which files becomes so much harder. It's an unnecessary change which already comes with more new problems than it fixes. This script doesn't change that often after all.

File: buildme.pl (hunk 3 above)
Comment (fsbruva) [reply buildme.pl]: Change made.

**Comment by fsbruva** [comment]
1. The script has guaranteed obsolescence with the existing Perl version scheme, when we KNOW certain config will likely never be built on newer Perl. 
2. I used the variables because *I* couldn't figure out what files each configuration used. 
3. Using the variables approach, adding aarch64 support would have changed 3 lines of code instead of 11. Less opportunities to introduce errors.

**Comment by fsbruva** [comment]
@mherger Did the second rebase address your concerns? I added additional comments, and tried to improve the readability by grouping the variable declarations a little differently.

**Comment by mherger** [comment]
As I mentioned previously I'd prefer to have a static list. Yes, it's more cumbersome to update. But what you get is what you see. With this tons of nested variables it's almost impossible to figure out what files a specific build would include or not.

**Comment by fsbruva** [comment]
I understand your concerns, especially with nested variables. 
Two more questions:
1. Do you see any value in the future Perl version compatibility stuff?
2. Are you open to taking advantage of the pattern matching capabilities of grep? 
"`\-linux`"
returns the same list as 
"`i386-linux x86_64-linux sparc-linux arm-linux armhf-linux powerpc-linux aarch64-linux`"

**Comment by mherger** [comment]
I don't see why we should spend that much time on those few lines. There's so little change over time, why bother? I prefer to have it human readable than to have it the most elegant/flexible.

**Pull request closed without merging**